ext/ddtrace_profiling_native_extension/collectors_cpu_and_wall_time_worker.c

```diff
@@ -363,11 +363,17 @@ static void *run_sampling_trigger_loop(void *state_ptr) {
     state->stats.trigger_sample_attempts++;
 
     // TODO: This is still a placeholder for a more complex mechanism. In particular:
-    // * We want to signal a particular thread or threads, not the process in general
     // * We want to track if a signal landed on the thread holding the global VM lock and do something about it
     // * We want to do more than having a fixed sampling rate
 
-    kill(getpid(), SIGPROF);
+    current_gvl_owner owner = gvl_owner();
+    if (owner.valid) {
+      pthread_kill(owner.owner, SIGPROF);
```

Comment:
The note about the race condition would be welcome here as a comment.

Reply:
Added note in 67ebc43b7bbea706829ab1ad943f4ddc72b80408